tests/js/spec/views/issueList/organizationSavedSearchMenu.spec.jsx

```diff
@@ -26,58 +37,70 @@ describe('IssueListSavedSearchMenu', () => {
   ];
   const onSelect = jest.fn();
   const onDelete = jest.fn();
-  beforeEach(() => {
-    organization = TestStubs.Organization({access: ['org:write']});
-    wrapper = mountWithTheme(
+
+  function renderSavedSearch({organization} = {}) {
+    return mountWithTheme(
       <IssueListSavedSearchMenu
-        organization={organization}
+        organization={organization ?? TestStubs.Organization({access: ['org:write']})}
         savedSearchList={savedSearchList}
         onSavedSearchSelect={onSelect}
         onSavedSearchDelete={onDelete}
         query="is:unresolved assigned:lyn@sentry.io"
       />,
-      TestStubs.routerContext()
+      {context: TestStubs.routerContext()}
     );
-  });
+  }
 
   afterEach(() => {
-    MockApiClient.clearMockResponses();
     jest.resetAllMocks();
   });
 
-  describe('removing a saved search', () => {
-    it('shows a delete button with access', () => {
-      const button = wrapper.find('MenuItem').first().find('button[aria-label="delete"]');
-      expect(button).toHaveLength(1);
-    });
-
-    it('does not show a delete button without access', () => {
-      organization.access = [];
-      wrapper.setProps({organization});
+  it('shows a delete button with access', () => {
+    renderSavedSearch();
+    const assignedToMe = screen.getByText('Assigned to me');
+    expect(
+      within(assignedToMe.parentElement.parentElement).getByRole('button', {
+        name: 'delete',
+      })
+    ).toBeInTheDocument();
+  });
 
-      const button = wrapper.find('MenuItem').first().find('button[aria-label="delete"]');
-      expect(button.exists()).toBe(false);
-    });
+  it('does not show a delete button without access', () => {
+    renderSavedSearch({organization: TestStubs.Organization({access: []})});
+    const assignedToMe = screen.getByText('Assigned to me');
+    expect(
+      within(assignedToMe.parentElement.parentElement).queryByRole('button', {
+        name: 'delete',
+      })
```

Comment:
I'm wondering if there's a better way to do this check instead of using `parentElement.parentElement`

Reply:
switched to test id

Reply:
Thanks, @scttcper but I think the changed code was fine. I was referring to this part `assignedToMe.parentElement.parentElement`

Reply:
thanks! should we update here too?

Reply:
@priscilawebdev whoops! wow